File: Java/WoM_T3/src/main/java/sep3/service/interfaces/IContainerService.java
package sep3.service.interfaces;

import sep3.dto.ContainerDtos;

public interface IContainerService {

    // Create new container
    ContainerDtos.ContainerDto create(ContainerDtos.CreateContainerDto dto);

    // Update existing container
    ContainerDtos.ContainerDto update(ContainerDtos.UpdateContainerDto dto);

    // Delete by id
    void delete(long id);

    // Get single container by id
    ContainerDtos.ContainerDto get(long id);

    // Get all containers
    ContainerDtos.ContainerListDto getAll();
}

File: C_Sharp/Server/WoM_WebApi/Services/Interfaces/IContainerService.cs
// File: Server/WoM_WebApi/Services/Interfaces/IContainerService.cs
using ApiContracts;

namespace WoM_WebApi.Services.Interfaces;

// Service: containers CRUD
public interface IContainerService
{
    Task<ContainerDto> CreateAsync(CreateContainerDto dto); // create container
    Task<ContainerDto> GetByIdAsync(long id);               // get one
    Task<ContainerListDto> GetAllAsync();                   // list all
    Task<ContainerDto> UpdateAsync(long id, UpdateContainerDto dto); // update
    Task DeleteAsync(long id);                              // delete
}
